src/rs_distributions/distributions/rice.py

```diff
@@ -0,0 +1,215 @@
+import numpy as np
+import torch
+import math
+from torch import distributions as dist
+
+
+class RiceIRSample(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, nu, sigma, samples, dnu, dsigma):
+        ctx.save_for_backward(dnu, dsigma)
+        return samples
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        (
+            grad_nu,
+            grad_sigma,
+        ) = ctx.saved_tensors
+        return grad_output * grad_nu, grad_output * grad_sigma, None, None, None
+
+
+class Rice(dist.Distribution):
+    """
+    The Rice distribution is useful for modeling acentric structure factor amplitudes in
+    X-ray crystallography. It is the amplitude distribution corresponding to a bivariate
+    normal in the complex plane.
+
+    ```
+    x ~ MVN([ν, 0], σI)
+    y = sqrt(x[0] * x[0] + x[1] * x[1])
+    ```
+    The parameters ν and σ represent the location and standard deviation of an isotropic, bivariate normal.
+    If x is drawn from the normal with location [ν, 0i] and covariance,
+    ```
+    | σ   0 |
+    | 0  σi |
+    ```
+    the distribution of amplitudes, `y = sqrt(x * conjugate(x))`, follows a Rician distribution.
+
+    Args:
+        nu (float or Tensor): location parameter of the underlying bivariate normal
+        sigma (float or Tensor): standard deviation of the underlying bivariate normal (must be positive)
+        validate_args (bool, optional): Whether to validate the arguments of the distribution.
+        Default is None.
+    """
+
+    arg_constraints = {
+        "nu": dist.constraints.nonnegative,
+        "sigma": dist.constraints.positive,
```

Comment:
Should sigma be nonnegative?

Reply:
Sigma must be greater than zero otherwise the distribution is a delta function.